src/polyswarm_api/__init__.py

```diff
@@ -735,58 +736,110 @@ def _reveal_closed(self, result):
         """
         return await self._delete_hunt(hunt_id, 'historical')
 
-    async def _get_hunt_results(self, hunt_id=None, scan_type='live'):
+    async def _get_hunt_results(self, hunt_id=None, scan_type='live', limit=MAX_HUNT_RESULTS, offset=0,
+                                all_results=False, with_metadata=False, with_bounties=False):
         """
 
         :param hunt_id: Rule ID (None if latest rule results are desired)
         :param scan_type: Type of scan, "live" or "historical"
+        :param limit: Limit the number of scan results, returns the most recent hits
+        :param offset: Offset into the result set to return
+        :param all_results: Boolean on whether to fetch all results. Note: this ignores limit/offset and can take awhile.
+        :param with_metadata: Whether to include metadata in the results
+        :param with_bounties: Whether to include bounty results in the results
         :return: Matches to the rules
         """
 
-        params = {}
+        if limit > MAX_HUNT_RESULTS:
+            limit = MAX_HUNT_RESULTS
```

Comment:
Maybe log something here to let the user know that the value he provided is not being respected?

Reply:
Agreed, good catch.